### PR TITLE
fix: Using named capture groups for the regex

### PR DESCRIPTION
**Pull request by krackerman**

### CODE REVIEW

File: src/logic/process-logs.js [hunk 1]
```diff
@@ -8,14 +8,15 @@ const loggly = require('./services/loggly');
...
+  /^REPORT RequestId: (?<requestId>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\t/,
+  /Duration: (?<duration>\d+.\d+) ms\t/,
+  /Billed Duration: (?<billedDuration>\d+) ms\s?\t/,
+  /Memory Size: (?<maxMemory>\d+) MB\t/,
+  /Max Memory Used: (?<memory>\d+) MB\t/,
+  /(?:Init Duration: (?<initDuration>\d+.\d+) ms\t)?\n/,
   // https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html
-  /(?:XRAY TraceId: (\d+-[0-9a-f]{8}-[0-9a-f]{24})\tSegmentId: ([0-9a-f]{16})\tSampled: (true|false)\t\n)?$/
+  // eslint-disable-next-line max-len
+  /(?:XRAY TraceId: (?<traceId>\d+-[0-9a-f]{8}-[0-9a-f]{24})\tSegmentId: (?<segmentId>[0-9a-f]{16})\tSampled: (?<sampled>true|false)\t\n)?$/
```
Comment (krackerman) [inline src/logic/process-logs.js]: Should we split this line up?

Comment (simlu) [reply src/logic/process-logs.js]: yes please

Comment (MacMcIrish) [reply src/logic/process-logs.js]: We want to make it clear that this whole group is considered optional, so when you split this up @krackerman , do it like:
```suggestion
  ...[
    /(?:XRAY TraceId: (?<traceId>\d+-[0-9a-f]{8}-[0-9a-f]{24})\t/,
    /SegmentId: (?<segmentId>[0-9a-f]{16})\t/,
    /...\n)?/
  ]
```

File: src/logic/process-logs.js [hunk 2]
```diff
@@ -8,14 +8,16 @@ const loggly = require('./services/loggly');
...
-  /Duration: (\d+.\d+) ms\t/,
-  /Billed Duration: (\d+) ms\s?\t/,
-  /Memory Size: (\d+) MB\t/,
-  /Max Memory Used: (\d+) MB\t/,
-  /(?:Init Duration: (\d+.\d+) ms\t)?\n/,
+  /^REPORT RequestId: (?<requestId>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\t/,
+  /Duration: (?<duration>\d+.\d+) ms\t/,
+  /Billed Duration: (?<billedDuration>\d+) ms\t/,
+  /Memory Size: (?<maxMemory>\d+) MB\t/,
+  /Max Memory Used: (?<memory>\d+) MB\t/,
```
Comment (MacMcIrish) [inline src/logic/process-logs.js]: Fix

**Pull request merged**